boot/freeldr/freeldr/CMakeLists.txt

```diff
@@ -29,10 +29,12 @@ list(APPEND FREELDR_ARC_SOURCE
     arcname.c
     arch/arcemul.c
     arch/archwsup.c
+    arch/i386/halstub.c
+    arch/i386/ntoskrnl.c
```

Comment:
a matter of taste regarding the sorting. Consider moving those 2 lines at the very end of the list, since they are one level deeper than the rest. That would be more stringent, because you also places arch/arcemul.c after and not before arcname.c.